### ISSUE TITLE
Add an option to remove unphysical solutions data in the result object

### ISSUE THREAD

**Issue by jdelpino** [NonlinearOscillations/HarmonicBalance.jl#111]
Currently, unphysical branches are encoded with NaNs, but data about unphysical branches are kept. This is done in favor of plotting since NaNs are ignored. However, in problems where neither plotting (nor branch ordering) is important, storing unphysical branches seems an overkill. An option to sparsify data, in order to avoid storage of result objects filled with NaNs would be appreciated.

**Comment by oameye** [comment]
I agree. I already had I lot of annoyances with this. I propose we add a keyword to the `transform_solutions` function to filter the solutions of specific classes.

**Comment by jdelpino** [comment]
We should save by default physical solutions. We could then use simple Run-Length encoding to store the data:
[https://en.wikipedia.org/wiki/Run-length_encoding]
